Mathlib/Order/SuccPred/Basic.lean

```diff
@@ -424,6 +451,30 @@ section NoMaxOrder
 
 variable [NoMaxOrder α]
 
+@[simp]
+theorem lt_succ_iff : a < succ b ↔ a ≤ b :=
+  lt_succ_iff_of_not_isMax <| not_isMax b
+
+theorem succ_le_succ_iff : succ a ≤ succ b ↔ a ≤ b := by simp
+
+theorem succ_lt_succ_iff : succ a < succ b ↔ a < b := by simp
+
+alias ⟨le_of_succ_le_succ, _⟩ := succ_le_succ_iff
+
+alias ⟨lt_of_succ_lt_succ, _⟩ := succ_lt_succ_iff
+
+@[simp]
+theorem Iio_succ (a : α) : Iio (succ a) = Iic a :=
+  Iio_succ_of_not_isMax <| not_isMax _
+
+@[simp]
+theorem Ico_succ_right (a b : α) : Ico a (succ b) = Icc a b :=
+  Ico_succ_right_of_not_isMax <| not_isMax _
+
+@[simp]
+theorem Ioo_succ_right (a b : α) : Ioo a (succ b) = Ioc a b :=
+  Ioo_succ_right_of_not_isMax <| not_isMax _
```

Comment:
This is still true in a non-linear succ-archimedean order. Mind writing a TODO?

Reply:
Like so?